### PR TITLE
updated dependencies with latest versions of http4s and cirius

### PR DESCRIPTION
**Pull request by rajcspsg**
@vlovgr I got to know about cirius when I was learning cats and doing prototype.
I did library version update for this example since some of them are bit older. 
Please review and let me know your comments.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -22,17 +23,17 @@ libraryDependencies ++= Seq(
...
   "is.cir" %% "ciris-enumeratum",
   "is.cir" %% "ciris-refined"
-).map(_ % "0.9.2")
+).map(_ % "0.12.1")
 
 libraryDependencies ++= Seq(
   "org.http4s" %% "http4s-dsl",
   "org.http4s" %% "http4s-blaze-server"
-).map(_ % "0.18.5")
+).map(_ % "0.19.0")
```
[the remark this answers is not in the document]
Comment (rajcspsg) [reply build.sbt]: updated

File: src/main/scala/is/cir/example/application/Http4sApi.scala [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
+import org.http4s.{AuthScheme, Credentials, HttpApp, HttpRoutes, Request}
 import scala.concurrent.ExecutionContext
-import scala.concurrent.duration.{Duration, FiniteDuration}
+import scala.concurrent.duration._
+
+final case class Http4sApi()(implicit executionContext: ExecutionContext) extends HttpApiAlg[IO] {
+
+  private implicit val cs: ContextShift[IO] = IO.contextShift(executionContext)
+  private implicit val timer: Timer[IO] = IO.timer(executionContext)
+  private implicit val ioConcurrent: Concurrent[IO] = cats.effect.IO.ioConcurrentEffect
```
[the remark this answers is not in the document]
Comment (rajcspsg) [reply src/main/scala/is/cir/example/application/Http4sApi.scala]: @vlovgr updated

File: src/main/scala/is/cir/example/application/Http4sApi.scala [hunk 3]
```diff
@@ -32,34 +33,33 @@ final case class Http4sApi()(
         case _                                                        => false
       }
 
-  def withTimeout(timeout: Duration)(service: HttpService[IO]): HttpService[IO] =
-    timeout match {
-      case finite: FiniteDuration => Timeout(finite)(service)
-      case _                      => service
-    }
+  def withTimeout(timeout: Duration, logger: LoggingAlg[IO])(routes: HttpRoutes[IO]): HttpApp[IO] =
```
[the remark this answers is not in the document]
Comment (rajcspsg) [reply src/main/scala/is/cir/example/application/Http4sApi.scala]: removed

File: src/main/scala/is/cir/example/domain/ConfigAlg.scala [hunk 4]
```diff
@@ -69,9 +69,9 @@ abstract class ConfigAlg[F[_]](implicit me: MonadError[F, Throwable]) {
           }
       }
 
-    errorsOrConfig.flatMap {
+    errorsOrConfig.result flatMap  {
```
[the remark this answers is not in the document]
Comment (rajcspsg) [reply src/main/scala/is/cir/example/domain/ConfigAlg.scala]: updated

File: src/main/scala/is/cir/example/domain/config/Config.scala [hunk 5]
```diff
@@ -23,18 +19,18 @@ final case class Config(
...
-      Show.fromToString
+  import cats.implicits.catsStdShowForDuration
+  implicit def showNonEmptyString: Show[NonEmptyString] = Show.fromToString
+  import ApiConfig.{showApiConfig, showEnumEntry }
+
+  implicit val showConfig: Show[Config] = {
 
-    implicit def showEnumEntry[E <: EnumEntry]: Show[E] =
-      Show.show(_.entryName)
+    implicitly[Show[ApiConfig]]
```
[the remark this answers is not in the document]
Comment (rajcspsg) [reply src/main/scala/is/cir/example/domain/config/Config.scala]: removed

File: src/main/scala/is/cir/example/domain/config/Config.scala [hunk 6]
```diff
@@ -23,18 +19,18 @@ final case class Config(
...
 
 object Config {
-  implicit val showConfig: Show[Config] = {
-    implicit val showDuration: Show[Duration] =
-      Show.fromToString
 
-    implicit val showInetAddress: Show[InetAddress] =
-      Show.fromToString
+  import cats.implicits.catsStdShowForDuration
+  implicit def showNonEmptyString: Show[NonEmptyString] = Show.fromToString
```
[the remark this answers is not in the document]
Comment (rajcspsg) [reply src/main/scala/is/cir/example/domain/config/Config.scala]: @vlovgr I tried to use implicit Show instance. I ran into https://github.com/fthomas/refined/issues/553 .
Shall we keep it `implicit def showNonEmptyString: Show[NonEmptyString] = Show.fromToString`  till new release of refined-cats.

[the remark this answers is not in the document]
Comment (vlovgr) [reply src/main/scala/is/cir/example/domain/config/Config.scala]: Yes, lets leave it for now.

**Comment by vlovgr** [comment]
Thanks again @rajcspsg!

**Comment by rajcspsg** [comment]
Thanks @vlovgr for accepting my PR :)

**Pull request merged**